src/schemata/definitions/workflow-template/action-objects/update-resource.yml

```
@@ -21,6 +22,24 @@ additionalProperties:
             - workflow_run.data.id
             - deviceField.id
     - oneOf:
+        - type: object
+          required:
+            - type
+            - attributes
+          properties:
+            type:
+              type: string
+              enum: [device]
+            attributes:
+              type: object
+              additionalProperties: false
+              properties:
+                custom_attributes:
+                  $ref: '#/definitions/script'
+                  description: |
+                    To update a custom attribute, the script must resolve to a hash object with `name`, `type` and `value` attributes. The supported custom attribute data types are `number`, `text`, `boolean` (checkbox), `date`, `datetime`, `file`, `image`, `physicalQuantity`, `relation`, and `duration`.
```

```suggestion
                  description: &custom_attributes_description |
                    To update a custom attribute, the script must resolve to a hash object with `name`, `type` and `value` attributes. The supported custom attribute data types are `number`, `text`, `boolean` (checkbox), `date`, `datetime`, `file`, `image`, `physicalQuantity`, `relation`, and `duration`.
```
let's write this once and use that anchor throughout this file to make sure we are consistent

---

actually we can move this up one level, right?

```yml
custom_attributes: &custom_attributes
  # rest of definition
```

---

hmm could you please reference the doc how to use the ampersand as the anchor, I have not seen this syntax before 😳

---

[Anchors and Aliases in the official docs](https://yaml.org/spec/1.2-old/spec.html#id2765878), which seems not very helpful 🥴 . Here's [another google](https://support.atlassian.com/bitbucket-cloud/docs/yaml-anchors/) result with examples that looks better.

The idea is that you mark an anchor with `&anchor_name` and then use that alias to reference it elsewhere with `*anchor_name`. That way we don't have to repeat ourselves so many times in this file.

I'm not 100% sure if we've used this before here and if it plays nice with the parser we use, but I would expect it does. The generated output in the JSON files should look the same as when typing it out everywhere, then it works.

---

like this:

```yml
              properties:
                custom_attributes: &custom_attributes
                  $ref: '#/definitions/script'
                  description: |
                    To update a custom attribute, the script must resolve to a hash object with `name`, `type` and `value` attributes. The supported custom attribute data types are `number`, `text`, `boolean` (checkbox), `date`, `datetime`, `file`, `image`, `physicalQuantity`, `relation`, and `duration`.
                name:
                  $ref: '#/definitions/script'
        - type: object
          required:
            - type
            - attributes
          properties:
            type:
              type: string
              enum: [workflow_run]
            attributes:
              type: object
              additionalProperties: false
              properties:
                custom_attributes: *custom_attributes
                title:
                  $ref: '#/definitions/script'
```

---

ahh it is yaml specification, not the json schema, thank you!